### ISSUE TITLE
clippy is not part of the build/test

### ISSUE THREAD

**Issue by yegor256** [objectionary/sodg#76]
When I run `cargo build` or `cargo test`, the `clippy` is not running. I have to do it with an extra step: `cargo clippy`. Would be great to make this style checker part of the build.

**Comment by yegor256** [comment]
@UARTman please, help

**Comment by l3r8yJ** [comment]
@yegor256 can i take this one?

**Comment by yegor256** [comment]
@l3r8yJ yes, please

**Comment by l3r8yJ** [comment]
@yegor256 I can make clippy as annoying as possible, do we need it?

**Comment by l3r8yJ** [comment]
@yegor256 take a look, please 

Here's [what](https://pastebin.com/7xGWWPta) looks like a master build with the most complicated settings, can you suggest removing some of them?

**Comment by l3r8yJ** [comment]
also, we can add @0pdd to the project and I will create puzzles for each violation, what do you think?

**Comment by yegor256** [comment]
@l3r8yJ yes, of course, let's turn clippy to its maximum power AND make it part of the build in Rultor and in GHA. @0pdd is already in the repo.

**Comment by yegor256** [comment]
@l3r8yJ thanks
